resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -71,17 +75,64 @@ private[spark] object KubernetesUtils extends Logging {
    * - File URIs with scheme local:// resolve to just the path of the URI.
    * - Otherwise, the URIs are returned as-is.
    */
-  def resolveFileUrisAndPath(fileUris: Iterable[String]): Iterable[String] = {
+  def resolveFileUrisAndPath(fileUris: Iterable[String], conf: Option[SparkConf] = None)
+  : Iterable[String] = {
     fileUris.map { uri =>
-      resolveFileUri(uri)
+      resolveFileUri(uri, conf)
     }
   }
 
-  def resolveFileUri(uri: String): String = {
+  /**
+   * Get the final path for a client file, if not return the uri as is.
+   *
+   */
+  def getDestPathIfClientFile(uri: String, conf: SparkConf): String = {
+    val fileUri = Utils.resolveURI(uri)
+    val fileScheme = Option(fileUri.getScheme).getOrElse("file")
+    if (fileScheme == "client") {
+      if (conf.get(KUBERNETES_FILE_UPLOAD_PATH).isDefined) {
+        val uploadPath = conf.get(KUBERNETES_FILE_UPLOAD_PATH).get
+        s"${uploadPath}/${fileUri.getPath.split("/").last}"
```

Comment:
So a file `client://path/to/app1.jar` will be uploaded to `${uploadPath}/app1.jar`? What if two client-local files at different local paths have the same file name?

Reply:
I m not supporting that. I thought of that. People should make sure dont create a conflict. Otherwise will have to create random paths.

Reply:
Please document this clearly, i.e., all client-side dependencies will be uploaded to the given path with a flat directory structure.

Reply:
will do thanx.